modules/apps/commerce/commerce-product-service/src/main/java/com/liferay/commerce/product/internal/model/listener/AccountEntryModelListener.java

```diff
@@ -35,6 +41,29 @@
 @Component(service = ModelListener.class)
 public class AccountEntryModelListener extends BaseModelListener<AccountEntry> {
 
+	@Override
+	public void onAfterRemove(AccountEntry accountEntry) {
+		List<CommerceCatalog> commerceCatalogs =
+			_commerceCatalogLocalService.getCommerceCatalogs(
+				accountEntry.getCompanyId(), false);
+
+		for (CommerceCatalog commerceCatalog : commerceCatalogs) {
+			commerceCatalog.setAccountEntryId(0);
```

Comment:
You should update only the catalogs that are linked to the deleted account, not every catalog of the company